treeherder/model/models.py

```diff
@@ -168,7 +168,7 @@ class Commit(models.Model):
     A single commit in a push
     '''
     push = models.ForeignKey(Push, on_delete=models.CASCADE, related_name='commits')
-    revision = models.CharField(max_length=40)
+    revision = models.CharField(max_length=40, db_index=True)
```

Comment:
I'm adding this to speed up when we can't find a `Push` with the right `revision`.  This will at least given them something to go on.  But there was no index on this field for searching.  So, Voila!